backends/ebpf/psa/README.md

```diff
@@ -164,6 +164,103 @@ a new element to the outer map (indexed by clone session or multicast group iden
 or `multicast_group` in a PSA program) and initialize an inner map. To add a new clone session/multicast group member,
 a con1trol plane must add new element to the inner map.
 
+## P4 match kinds
+
+The PSA-eBPF compiler currently supports the following P4 match kinds: `exact`, `lpm`, `ternary`.
+
+### exact
+
+An `exact` table is implemented using the BPF hash map. A P4 table is considered an `exact` table if all its match fields are defined as `exact`.
+Then, the PSA-eBPF compiler generates a BPF hash map instance for each P4 table instance. The hash map key as a concatenation of P4 match fields translated to eBPF representation.
+Each `apply()` operation is translated into a lookup to the BPF hash map. The value is used to determine an action and its parameters. 
+
+### lpm
+
+An `lpm` table is implemented using the BPF `LPM_TRIE` map. A P4 table is considered an `lpm` table if it contains a single `lpm` field and no `ternary` fields.
+The PSA-eBPF compiler generates a BPF `LPM_TRIE` map instance for each P4 table instance. The hash map key as a concatenation of P4 match fields translated to eBPF representation.
+Moreover, the PSA-eBPF compiler shuffles the match fields and places the `lpm` field in the last position. Each `apply()` operation is translated into a lookup to the `LPM_TRIE` map.
```

Comment:
This may be a problem since the control-plane has to know this. For example, if you plan p4Runtime support, I am not sure it will be easy. (Perhaps it is, just shuffle the data from the control-plane in the same way).
Wouldn't it be simpler to just require users to put the field last and reject other programs?

Reply:
Shuffling the data from control plane shouldn't be a problem, since we have P4Info. IDK we might want to revisit that later (once we start integrating PSA-eBPF with a P4Runtime stack), but I'd keep it as is, for now.